libraries/standard/http/src/http_client.c

```diff
@@ -1198,50 +1430,83 @@ static HTTPStatus_t findHeaderInResponse( const uint8_t * pBuffer,
 
         /* Header is not present in buffer. */
         LogWarn( ( "Header not found in response buffer: "
-                   "RequestedHeader=%.*s", fieldLen, pField ) );
+                   "RequestedHeader=%.*s",
+                   ( int ) fieldLen,
+                   pField ) );
 
         returnStatus = HTTP_HEADER_NOT_FOUND;
     }
     else if( context.valueFound == 0u )
     {
         /* The response buffer is invalid as only the header field was found
          * in the "<field>: <value>\r\n" format of an HTTP header. */
+
+        /* MISRA rule 10.5 notes casting an unsigned integer to an enum type. This
+         * violation is suppressed because the http-parser library directly sets
+         * and maps http_parser.http_errno to the enum http_errno. */
+        /* coverity[misra_c_2012_rule_10_5_violation] */
         LogError( ( "Unable to find header value in response: "
                     "Response data is invalid: "
                     "RequestedHeader=%.*s, ParserError=%s",
-                    fieldLen, pField,
+                    ( int ) fieldLen,
+                    pField,
                     http_errno_description( HTTP_PARSER_ERRNO( &( parser ) ) ) ) );
         returnStatus = HTTP_INVALID_RESPONSE;
     }
     else
     {
-        /* Empty else (when assert and logging is disabled) for MISRA 15.7 compliance. */
+        /* Empty else (when assert and logging is disabled) for MISRA 15.7
+         * compliance. */
 
         /* Header is found. */
         assert( ( context.fieldFound == 1u ) && ( context.valueFound == 1u ) );
 
         LogDebug( ( "Found requested header in response: "
                     "HeaderName=%.*s, HeaderValue=%.*s",
-                    fieldLen, pField,
-                    *pValueLen, *pValueLoc ) );
+                    ( int ) fieldLen,
+                    pField,
+                    ( int ) ( *pValueLen ),
+                    *pValueLoc ) );
     }
 
-    /* If the header field-value pair is found in response, then the return value of "on_header_value"
-     * callback (related to the header value) should cause the http_parser.http_errno to be "CB_header_value". */
+    /* If the header field-value pair is found in response, then the return
+     * value of "on_header_value" callback (related to the header value) should
+     * cause the http_parser.http_errno to be "CB_header_value". */
+
+    /* MISRA rule 10.5 notes casting an unsigned integer to an enum type. This
+     * violation is suppressed because the http-parser library directly sets
+     * and maps http_parser.http_errno to the enum http_errno. */
+    /* coverity[misra_c_2012_rule_10_5_violation] */
     if( ( returnStatus == HTTP_SUCCESS ) &&
-        ( ( parser.http_errno != HPE_CB_header_value ) ) )
-    {
-        LogError( ( "Header found in response but http-parser returned error: ParserError=%s",
+        ( ( ( ( enum http_errno ) ( parser.http_errno ) ) != HPE_CB_header_value ) ) )
```

Comment:
As the MISRA warning still persists despite the explicit cast, is the cast required anymore or is there a gcc warning that this cast addresses?

Reply:
Let get rid of this, tried stuff out then left it